crates/rattler_repodata_gateway/src/fetch/mod.rs

```diff
@@ -725,7 +725,10 @@ async fn stream_and_decode_to_file(
     // Clone the file handle and create a hashing writer so we can compute a hash
     // while the content is being written to disk.
     let file = tokio_fs::File::from_std(fs_err::File::from_parts(
-        temp_file.as_file().try_clone().unwrap(),
+        temp_file
+            .as_file()
+            .try_clone()
+            .map_err(FetchRepoDataError::IoError)?,
```

Comment:
Can this happen? Just wondering :)

Reply:
I guess the clone operation can fail if https://man7.org/linux/man-pages/man2/fcntl.2.html will return something unexpected which will be converted by https://docs.rs/cvt/latest/cvt/. 
don't know what are the `triggers` for this

Reply:
Same here! Use into_parts to forgo to the try_clone

Reply:
I think I cannot do it here.

```HashingWriter::<_, Blake2b256>::new(file);``` take ownership of the file. 
If I will do `temp_file.into_parts()` I will lose this file handle when creating it `from_parts` so later, I will not be able to create NamedTempFile again.